py3status/modules/mpris.py

```diff
@@ -0,0 +1,552 @@
+            'album': None,
+            'artist': None,
+            'error_occurred': False,
+            'length': None,
+            'player': None,
+            'state': STOPPED,
+            'title': None
+        }
+
+        if self._player is None:
+            return
+
+        try:
+            self._data['player'] = self._player.Identity
+            self._data['state'] = self._get_state()
+            self._update_metadata()
+        except Exception:
+            self._data['error_occurred'] = True
+
+    def _get_button_state(self, control_state):
+        try:
+            # Workaround: The last parameter returns True for the Stop button.
+            clickable = getattr(self._player, control_state['clickable'], True)
+        except Exception:
+            clickable = False
+
+        if control_state['action'] == 'Play' and self._data['state'] == PLAYING:
+            clickable = False
+        elif control_state['action'] == 'Pause' and (
+             self._data['state'] == STOPPED or self._data['state'] == PAUSED):
+            clickable = False
+        elif control_state['action'] == 'Stop' and self._data['state'] == STOPPED:
+            clickable = False
+
+        return clickable
+
+    def _get_state(self, player=None, state=None):
+        if state:
+            playback_status = state
+        elif player:
+            playback_status = player.PlaybackStatus
+        else:
+            playback_status = self._player.PlaybackStatus
+
+        if playback_status == 'Playing':
+            return PLAYING
+        elif playback_status == 'Paused':
+            return PAUSED
+        else:
+            return STOPPED
+
+    def _get_text(self):
+        """
+        Get the current metadata
+        """
+        if self._data['state'] == PLAYING:
+            color = self.py3.COLOR_PLAYING or self.py3.COLOR_GOOD
+            state_symbol = self.state_play
+        elif self._data['state'] == PAUSED:
+            color = self.py3.COLOR_PAUSED or self.py3.COLOR_DEGRADED
+            state_symbol = self.state_pause
+        else:
+            color = self.py3.COLOR_STOPPED or self.py3.COLOR_BAD
+            state_symbol = self.state_stop
+
+        if self._data['error_occurred']:
+            color = self.py3.COLOR_BAD
+
+        try:
+            ptime_ms = self._player.Position
+            ptime = _get_time_str(ptime_ms)
+        except Exception:
+            ptime = None
+
+        if '{time}' in self.format and self._data['state'] == PLAYING:
+            # Don't get trapped in aliasing errors!
+            update = time() + 0.5
+        else:
+            update = self.py3.CACHE_FOREVER
+
+        placeholders = {
+            'player': self._data['player'],
+            'state': state_symbol,
+            'album': self._data['album'],
+            'artist': self._data['artist'],
+            'length': self._data['length'],
+            'time': ptime,
+            'title': self._data['title'] or 'No Track',
```

Comment:
When we set `'artist'` and `'title'` we get an _invalid format_ error. Without this two fields the module runs like a dog behind a cat.

Is this an issue in the py3status code?